lib/mvi/responses/id_parser.rb

```diff
@@ -12,28 +12,35 @@ class IdParser
       # {source id}^{id type}^{assigning facility}^{assigning authority}^{id status}
       # id type - NI = national identifier
       #           PI = patient identifier
-      # id status -  A = active
-      #              P = permanent
-      #              H = deprecated(do not use)
-      #              L = pending local merge (do not use)
-      #              PCE = ??
+
+      # id status -  A = Active
+      #              P = Permanent
+
+      # Should definitely NOT be using id_status
+      #              H = Deprecated due to local merge
+      #              D = Deprecated from a Duplicate
+      #              M = Deprecated from a Mismatch
+      #              U - Deprecated from an Unlink
+      #              L = pending local merge - This is a value identified for correlations including EDIPI when the
+      #                   identifier has been deprecated/associated to another active ID.
+      #              PCE = Pending Cat Edit correlations (unsure if this should be used, likely not)
 
       def parse(ids)
         ids = ids.map(&:attributes)
-
+        # rubocop:disable LineLength
         {
-          icn: select_ids(select_extension(ids, ICN_REGEX, CORRELATION_ROOT_ID))&.first,
-          sec_id: select_ids(select_extension(ids, /^\w+\^PN\^200PROV\^USDVA\^\w+$/, CORRELATION_ROOT_ID))&.first,
-          mhv_ids: select_ids(select_extension(ids, /^\w+\^PI\^200MH.{0,1}\^\w+\^\w+$/, CORRELATION_ROOT_ID)),
-          active_mhv_ids: select_ids(select_extension(ids, /^\w+\^PI\^200MH.{0,1}\^\w+\^A$/, CORRELATION_ROOT_ID)),
-          edipi: select_ids(select_extension(ids, /^\w+\^NI\^200DOD\^USDOD\^\w+$/, EDIPI_ROOT_ID))&.first,
-          vba_corp_id: select_ids_except(select_extension(ids, /^\w+\^PI\^200CORP\^USVBA\^\w+$/, CORRELATION_ROOT_ID),
-                                         %w[H L])&.first,
+          icn:              select_ids(select_extension(ids, ICN_REGEX,                           CORRELATION_ROOT_ID))&.first,
+          sec_id:           select_ids(select_extension(ids, /^\w+\^PN\^200PROV\^USDVA\^\w+$/,    CORRELATION_ROOT_ID))&.first,
+          mhv_ids:          select_ids(select_extension(ids, /^\w+\^PI\^200MH.{0,1}\^\w+\^\w+$/,  CORRELATION_ROOT_ID)),
+          active_mhv_ids:   select_ids(select_extension(ids, /^\w+\^PI\^200MH.{0,1}\^\w+\^A$/,    CORRELATION_ROOT_ID)),
+          edipi:            select_ids(select_extension(ids, /^\w+\^NI\^200DOD\^USDOD\^\w+$/,     EDIPI_ROOT_ID))&.first,
+          vba_corp_id:      select_ids(select_extension(ids, /^\w+\^PI\^200CORP\^USVBA\^(A|P)$/,  CORRELATION_ROOT_ID))&.first,
```

Comment:
The `A or P` statuses are now checked inline here with that trailing regex. Seems consistent enough with the rest of this, though they are all a bit complicated to parse mentally (at least for me).

I'd love to see @jholton take a quick look at this since he did the work in the corresponding PR you mention and likely has the most context

👍 for deleting more code!

Reply:
I'm on it.  I haven't reviewed the PR yet, because I'm currently trying to get more information about these IDs (and statuses) from MVI engineers.